Tests/GRPCCodeGenTests/Internal/Translator/SnippetBasedTranslatorTests.swift

```diff
@@ -0,0 +1,200 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import XCTest
+
+@testable import GRPCCodeGen
+
+final class SnippetBasedTranslatorTests: XCTestCase {
+  typealias MethodDescriptor = GRPCCodeGen.CodeGenerationRequest.ServiceDescriptor.MethodDescriptor
+  typealias ServiceDescriptor = GRPCCodeGen.CodeGenerationRequest.ServiceDescriptor
+
+  let echoMethods: [MethodDescriptor] = {
+    var methods: [MethodDescriptor] = []
+    methods.append(
+      MethodDescriptor(
+        documentation: "Immediately returns an echo of a request.",
+        name: "Get",
+        isInputStreaming: false,
+        isOutputStreaming: false,
+        inputType: "Echo_EchoRequest",
+        outputType: "Echo_EchoResponse"
+      )
+    )
+    methods.append(
+      MethodDescriptor(
+        documentation: "Splits a request into words and returns each word in a stream of messages.",
+        name: "Expand",
+        isInputStreaming: false,
+        isOutputStreaming: true,
+        inputType: "Echo_EchoRequest",
+        outputType: "Echo_EchoResponse"
+      )
+    )
+    methods.append(
+      MethodDescriptor(
+        documentation:
+          "Collects a stream of messages and returns them concatenated when the caller closes.",
+        name: "Collect",
+        isInputStreaming: true,
+        isOutputStreaming: false,
+        inputType: "Echo_EchoRequest",
+        outputType: "Echo_EchoResponse"
+      )
+    )
+    methods.append(
+      MethodDescriptor(
+        documentation: "Streams back messages as they are received in an input stream.",
+        name: "Update",
+        isInputStreaming: true,
+        isOutputStreaming: true,
+        inputType: "Echo_EchoRequest",
+        outputType: "Echo_EchoResponse"
+      )
+    )
+    return methods
+  }()
+
+  var echoService: CodeGenerationRequest.ServiceDescriptor {
+    CodeGenerationRequest.ServiceDescriptor(
+      documentation: "An echo service.",
+      name: "Echo",
+      namespace: "echo",
+      methods: echoMethods
+    )
+  }
+
+  var codeGenerationRequest: CodeGenerationRequest {
+    CodeGenerationRequest(
+      fileName: "echo.grpc",
+      leadingTrivia: "Some really exciting license header 2023.",
+      dependencies: [],
+      services: [self.echoService],
+      lookupSerializer: {
+        "ProtobufSerializer<\($0)>()"
+      },
+      lookupDeserializer: {
+        "ProtobufDeserializer<\($0)>()"
+      }
+    )
+  }
+
+  func testTypealiasTranslate() throws {
```

Comment:
This test is much bigger than is necessary, it's also not sufficient as a test.

Here are a handful of tests I thought of and what each is checking.
- One service in a namespace with one method. ("basic" case)
- One service with an _empty_ namespace and one method. (Checks that an empty namespace is treated as no namespace)
- One service in a namespace with two methods. (Checks order of methods is preserved). 
- One service in a namespace with no methods. (A valid edge case)
- Two services in the same namespace each with no methods where the services are in reverse alphabetical order in the request. (Checks that the services are ordered alphabetically within a namespace)
- Two services, both with empty namespaces, each with no methods. (Checks that the services are ordered alphabetically even when there's no namespace.)
- Two services in different namespaces each with no methods where the namespaces are in reverse alphabetical order in the request. (Checks that the namespaces are ordered alphabetically)
- Two services, one in a namespace, one not in a namespace, each with no methods. (Checks the order of namespace vs. no namespace)

There's also a handful of additional validation tests we can do:
- Two services with the same name and no namespace throws an error
- Two services with the same name and same namespace throws an error
- Two methods with the same name in a service throws an error